src/app/kvm/components/KVMHeaderForms/AddLxd/AddLxd.test.tsx

```diff
@@ -60,115 +55,110 @@ describe("AddLxd", () => {
       }),
       zone: zoneStateFactory({
-        items: [zoneFactory()],
+        items: [zoneFactory({ id: 0 })],
       }),
     });
   });
 
   it("shows the credentials form by default", () => {
-    const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter
-          initialEntries={[{ pathname: "/kvm/add", key: "testKey" }]}
-        >
-          <CompatRouter>
-            <AddLxd clearSidePanelContent={jest.fn()} />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
-    );
+    renderWithBrowserRouter(<AddLxd clearSidePanelContent={jest.fn()} />, {
+      route: "/kvm/add",
+      state,
+    });
 
-    expect(wrapper.find("CredentialsForm").exists()).toBe(true);
-    expect(wrapper.find("AuthenticationForm").exists()).toBe(false);
-    expect(wrapper.find("SelectProjectForm").exists()).toBe(false);
+    expect(screen.getByText("Credentials")).toHaveClass("is-active");
+    expect(screen.getByText("Authentication")).not.toHaveClass("is-active");
+    expect(screen.getByText("Project selection")).not.toHaveClass("is-active");
   });
 
   it(`shows the authentication form if the user has generated a certificate for
-    the LXD KVM host`, () => {
+    the LXD KVM host`, async () => {
     const certificate = generatedCertificateFactory({
       CN: "my-favourite-kvm@host",
     });
+
     state.general.generatedCertificate.data = certificate;
-    const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter
-          initialEntries={[{ pathname: "/kvm/add", key: "testKey" }]}
-        >
-          <CompatRouter>
-            <AddLxd clearSidePanelContent={jest.fn()} />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
-    );
 
-    // Submit credentials form
-    submitFormikForm(wrapper, {
-      name: "my-favourite-kvm",
-      pool: 0,
-      power_address: "192.168.1.1",
-      zone: 0,
+    renderWithBrowserRouter(<AddLxd clearSidePanelContent={jest.fn()} />, {
+      route: "/kvm/add",
+      state,
     });
-    wrapper.update();
 
-    expect(wrapper.find("CredentialsForm").exists()).toBe(false);
-    expect(wrapper.find("AuthenticationForm").exists()).toBe(true);
-    expect(wrapper.find("SelectProjectForm").exists()).toBe(false);
+    // Submit credentials form
+    await userEvent.type(
+      screen.getByRole("textbox", { name: "Name" }),
+      "my-favourite-kvm"
+    );
+    await userEvent.selectOptions(
+      screen.getByRole("combobox", { name: "Resource pool" }),
+      "0"
+    );
+    await userEvent.selectOptions(
+      screen.getByRole("combobox", { name: "Zone" }),
+      "0"
+    );
+    await userEvent.type(
+      screen.getByRole("combobox", { name: "LXD address" }),
+      "192.168.1.1"
+    );
+    await userEvent.click(screen.getByRole("button", { name: "Next" }));
+
+    expect(screen.getByText("Credentials")).not.toHaveClass("is-active");
```

Comment:
This is too broad.  We need to be careful especially when checking for non-existance of elements.

It's too easy for this text to be displayed anywhere on the page and this may not mean what we intend (that the credentials form is not displayed).

You could add `aria-label="Credentials"` to the `CredentialsForm` and use it in the query.

Reply:
This isn't checking for the form but rather for the Stepper state - this uses list items, so I'll modify it to check for that role